numpy/distutils/fcompiler/__init__.py

```diff
@@ -465,8 +466,11 @@ def customize(self, dist = None):
         noarch = self.distutils_vars.get('noarch', noopt)
         debug = self.distutils_vars.get('debug', False)
 
-        f77 = self.command_vars.compiler_f77
-        f90 = self.command_vars.compiler_f90
+        # TODO: proper guards here and same for F90?
+        f77 = shlex.split(self.command_vars.compiler_f77,
```

Comment:
This seems to help from brute force debugging on the single test failure, but probably isn't right--we may for example just lose the `--coverage` flag completely by doing this instead of putting it in `extra_args` somewhere else.

I guess I'll have to drill down on that.

Reply:
Can you drop the `[0]`, and use `(compiler_fix=fix+fixflags+fflags)` instead of `(compiler_fix=[fix]+fixflags+fflags)` below?

Reply:
I tried it just now, but it breaks NumPy build:

```
  File "/Users/treddy/github_projects/numpy/numpy/distutils/fcompiler/__init__.py", line 524, in customize
    self.set_commands(compiler_fix=fix+fixflags+fflags)
TypeError: must be str, not list
```

Reply:
That's probably because you didn't drop the `[0]` on line 493

Reply:
That's not quite enough--the types of `f77` and `f90` variables have changed with indexing removal so I needed a few more edits locally.